pallet/src/tests/signer.rs

```diff
@@ -497,3 +583,71 @@ fn verify_old_multi_signer_requests_getting_removed() {
             );
         })
 }
+
+#[test]
+fn cheque_limit_in_multi_signer_execution_works() {
+    const BALANCE: Balance = 80_000_000_000_000;
+    let (bob_addr_32, bob_addr_mv) = addrs_from_ss58(BOB_ADDR).unwrap();
+    let (alice_addr_32, alice_addr_mv) = addrs_from_ss58(ALICE_ADDR).unwrap();
+    let (dave_addr_32, dave_addr_mv) = addrs_from_ss58(DAVE_ADDR).unwrap();
+    let (eve_addr_32, eve_addr_mv) = addrs_from_ss58(EVE_ADDR).unwrap();
+
+    ExtBuilder::default()
+        .with_balances(vec![
+            (bob_addr_32.clone(), BALANCE),
+            (alice_addr_32.clone(), BALANCE),
+            (dave_addr_32.clone(), BALANCE),
+            (eve_addr_32.clone(), BALANCE),
+        ])
+        .build()
+        .execute_with(|| {
+            // Roll to first block in case of block based event checkings and processes.
+            roll_to(1);
+
+            // Initialisation & Setup by developer Bob.
+            let module = assets::read_module_from_project("multiple-signers", "Dorm");
+            assert_ok!(MoveModule::publish_module(
+                RuntimeOrigin::signed(bob_addr_32.clone()),
+                module,
+                MAX_GAS_AMOUNT
+            ));
+            let script = assets::read_script_from_project("multiple-signers", "init_module");
+            let transaction_bc = script_transaction!(script, no_type_args!(), &bob_addr_mv);
+            assert_ok!(MoveModule::execute(
+                RuntimeOrigin::signed(bob_addr_32.clone()),
+                transaction_bc.clone(),
+                MAX_GAS_AMOUNT,
+                BALANCE,
+            ));
+
+            // Now only 2 of 3 planned signers will sign the script execution.
+            let script = assets::read_script_from_project("multiple-signers", "rent_apartment");
+            let transaction_bc = script_transaction!(
+                script,
+                no_type_args!(),
+                &alice_addr_mv,
+                &dave_addr_mv,
+                &eve_addr_mv,
+                &2u8
+            );
+            assert_ok!(MoveModule::execute(
+                RuntimeOrigin::signed(dave_addr_32.clone()),
+                transaction_bc.clone(),
+                MAX_GAS_AMOUNT,
+                BALANCE,
+            ));
+            assert_ok!(MoveModule::execute(
+                RuntimeOrigin::signed(eve_addr_32.clone()),
+                transaction_bc.clone(),
+                MAX_GAS_AMOUNT,
+                BALANCE,
+            ));
+            let res = MoveModule::execute(
+                RuntimeOrigin::signed(alice_addr_32.clone()),
+                transaction_bc.clone(),
+                MAX_GAS_AMOUNT,
+                BALANCE / 2,
+            );
+            assert!(verify_module_error_with_msg(res, "Aborted").unwrap());
```

Comment:
Could we add a comment on the reason behind this failure?

Is everybody expected to provide the exact BALANCE amount for the cheque_amount?

Reply:
Done